Extension/src/LanguageServer/extension.ts

```diff
@@ -194,6 +218,141 @@ function onDidChangeVisibleTextEditors(editors: vscode.TextEditor[]): void {
+function downloadFileToDestination(urlString, destinationPath: string, headers?: OutgoingHttpHeaders): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+        let parsedUrl: url.Url = url.parse(urlString);
+        let request: ClientRequest = https.request({
+            host: parsedUrl.host,
+            path: parsedUrl.path,
+            agent: util.getHttpsProxyAgent(),
+            rejectUnauthorized: vscode.workspace.getConfiguration().get("http.proxyStrictSSL", true),
+            headers: headers
+        }, (response) => {
+            if (response.statusCode === 301 || response.statusCode === 302) { // If redirected
+                // Download from new location
+                let redirectUrl: string | string[];
+                if (typeof response.headers.location === "string") {
+                    redirectUrl = response.headers.location;
+                } else {
+                    redirectUrl = response.headers.location[0];
+                }
+                return resolve(downloadFileToDestination(redirectUrl, destinationPath));
+            }
+            if (response.statusCode !== 200) { // If request is not successful
+                return reject();
+            }
+            // Write file using downloaded data
+            let downloadedBytes = 0; // tslint:disable-line
+            let createdFile: fs.WriteStream = fs.createWriteStream(destinationPath);
+            response.on('data', (data) => { downloadedBytes += data.length; });
+            response.on('end', () => { createdFile.close(); });
+            createdFile.on('close', () => { resolve(); this.updateSettingsAsync(); });
+            response.on('error', (error) => { reject(); });
+            response.pipe(createdFile, { end: false });
+        });
+        request.on('error', (error) => { reject(); });
+        request.end();
+    });
+}
+
+// TODO factor this out?
+async function parseJsonAtPath(path: string): Promise<any> {
+    try {
+        const exists: boolean = await util.checkFileExists(path);
+        if (exists) {
+            const fileContent: string = await util.readFileText(path);
+            return JSON.parse(fileContent);
+        }
+    } catch (error) {
+        console.log(error);
+    }
+}
+
+async function checkAndApplyUpdate(): Promise<void> {
+    // Get current version name, stripping out everything past (and incl.) the dash if present
+    // Return if the user is using a non-Release or non-Insider build (e.g. build ending in "-master")
+    const version: string = function(): string {
+        let version: string = util.packageJson["version"];
+        const dashOffset: number = version.indexOf("-");
+        if (dashOffset !== -1) {
+            if (version.substr(dashOffset + 1) !== "insiders") {
+                return null;
+            }
+            version = version.substr(0, dashOffset); // Strip out the suffix
+        }
+        return version;
+    }();
```

Comment:
We tend not to use a lot of IIFE's in our extension.  That's not saying you can't, but for something as complex as version management, it might be better to write up a class and have named functions doing the stuff you need.  It makes the code easier to maintain without requiring lots of comments.  It's my personal preference to avoid using them.